packages/xdl/src/Project.ts

```diff
@@ -1722,7 +1722,18 @@ export async function startReactNativeServerAsync(
     port: packagerPort,
     customLogReporterPath,
     assetExts: ['ttf'],
-    sourceExts: ['expo.js', 'expo.ts', 'expo.tsx', 'expo.json', 'js', 'json', 'ts', 'tsx'],
```

Comment:
Nice job, was planning on unifying these by merging the logic into this one function but couldn't find a way to emulate the ios/android platform extensions like we do on web.
```js
import { getManagedExtensions } from '@expo/config/build/paths';
const sourceExts  = getManagedExtensions([]);
/*
  "expo.ts",
  "expo.tsx",
  "expo.mjs",
  "expo.js",
  "expo.jsx",
  "ts",
  "tsx",
  "mjs",
  "js",
  "jsx",
  "json",
  "wasm",
*/
```

Reply:
I'm also thinking this order looks incorrect, will fix.

Reply:
Like this https://github.com/expo/expo-cli/pull/1187